src/utils.rs

```diff
@@ -151,15 +150,27 @@ pub(crate) fn positive_bn_random_from_transcript(
     }
 }
 
-/// Generate a random BigNumber in the range 1..N-1 (Z_N^*) (non-zero)
-pub(crate) fn random_bn_in_z_star<R: RngCore + CryptoRng>(rng: &mut R, n: &BigNumber) -> BigNumber {
-    for _ in 0..CRYPTOGRAPHIC_RETRY_MAX {
-        let bn = BigNumber::from_rng(n, rng);
-        if bn != BigNumber::zero() {
-            return bn;
-        }
-    }
-    BigNumber::zero()
+/// Generate a random `BigNumber` that is in the multiplicative group of integers modulo `n`.
+///
+/// This method draws a random element in the range `[0, n-1]`, then checks that it is
+/// non-zero and coprime with `n`.
+/// Coprimality is tested by checking that the least common multiple is equal to the product of
+/// `n` and the element.
+///
```

Comment:
I added this because I wanted to be explicit about how we're checking group membership, but actually it might be dumb to have 4 lines of documentation explaining what a 4-line function does.

Reply:
Why not just check the gcd?

Reply:
You know, I actually started with a gcd check and then second-guessed myself repeatedly to end up here. I will switch back.

Reply:
My guess is that if you look at how lcm is implemented, it calculates the gcd.